src_c/event.c

```diff
@@ -1284,14 +1284,21 @@ pg_event_dealloc(PyObject *self)
 }
 
 #ifdef PYPY_VERSION
-/* Because pypy does not work with the __dict__ tp_dictoffset. */
+/* Because in pypy PyObject_GenericGetAttr and PyObject_GenericSetAttr does
+ * not work with the __dict__ tp_dictoffset. */
 PyObject *
 pg_EventGetAttr(PyObject *o, PyObject *attr_name)
 {
-    /* Try e->dict first, if not try the generic attribute. */
-    PyObject *result = PyDict_GetItem(((pgEventObject *)o)->dict, attr_name);
+    /* check attribute first and if it cannot find it, check event.dict */
+    PyObject *result = PyObject_GenericGetAttr(o, attr_name);
+
     if (!result) {
-        return PyObject_GenericGetAttr(o, attr_name);
+        result = PyDict_GetItem(((pgEventObject *)o)->dict, attr_name);
```

Comment:
Since you changed the order of `PyObject_GenericGetAttr` and `PyDict_GetItem`, there is a new potential bug. 

While `PyDict_GetItem` is documented to not setting new errors, it is also documented that this can silencing errors in some cases. In this case, the error set by `PyObject_GenericGetAttr` is discarded, and this function will end up returning `NULL` without an error set.

Just to be safe, we should be using [`PyErr_Fetch`](https://docs.python.org/3/c-api/exceptions.html#c.PyErr_Fetch)/[`PyErr_Restore`](https://docs.python.org/3/c-api/exceptions.html#c.PyErr_Restore) pair around these bits

Reply:
Good catch!